cli/src/commands/prev.rs

```diff
@@ -78,25 +77,17 @@ pub(crate) fn cmd_prev(
             .heads()
             .contains(current_wc_id);
     let current_wc = workspace_command.repo().store().get_commit(current_wc_id)?;
-    let start_id = if edit {
-        current_wc_id
+    let offset = if current_wc.is_discardable() && !args.edit {
+        // If the current working copy is discardable and the user didn't
+        // specify --edit, then we move one commit further back. This is so that
+        // the new empty working commit is created on the parent of the current
+        // on (or further back depending on the value of args.amount). If we
+        // didn't do this, then we wouldn't move far enough.
+        args.offset + 1
```

Comment:
I still disagree with your model. The commits should be treated equally, even if they're discardable.

Reply:
I changed the PR to treat all commits equally and fix what is clearly a real issue (there is no way to create a new commit on `@-` with `jj prev`).

I disagree that commits should be treated equally, because doing so is often not practical from a usability perspective. That's why discardable commits are discarded when `@` moves. I have to admit that I really don't understand why some people are adamant that discardable commits are equal to all other commits. It seems like an argument about theoretical purity to me. I'm more interested in making the CLI easy to use.